src/routes/api/act.ts

```diff
@@ -55,25 +61,166 @@ router.get("/TWCodeCheck", async (req, res) => {
+        pnid: z.string().min(6, { message: "Pretendo Network IDs cannot be shorted than 6 characters!" }).max(16, { message: "Pretendo Network IDs cannot be longer than 16 characters!" }),
+    });
+
+    // gets the required data from the html form
+    const email = req.body.email; // User's email
+    const passwd = req.body.password; // Users plaintext password (Will be hashed)
+    const miiName = req.body.nickname; // User's Mii name/Mii nickname
+    const pid = req.body.principal_id; // User's principal id
+    const pnid = req.body.id; // The users Pretendo Network ID
+    const time = dayjs().format("HH:mm:ss"); // 24 hr time
+
+    if (!email || !passwd || !miiName || !pid || !pnid) {
+        const missingParams = ["email", "passwd", "miiName", "pid", "pnid"].filter(param => !eval(param));
+
+        const missingMessage = missingParams.join(", ").replace(/, ([^,]*)$/, " and $1");
+
+        res.status(400).json({
+            endpoint: "createAccount",
+            time: time,
+            error: 1,
+            result: {
+                message: `Missing required parameters: ${missingMessage}.`,
+            },
+        });
+        return;
+    }
+
+    // hashes a the password
+    const encryptPassword = async (pass: string) => {
+        const saltRounds = 10;
+        return await bcrypt.hash(pass, saltRounds);
+    };
+
+    // gets a json web token for that account
+    const getAccountJwt = (user: object) => {
+        return jwt.sign({ user }, JWT_SECRET, { expiresIn: "1h" });
+    };
+    
+    try {
+        const hashedPasswd = await encryptPassword(passwd);
+        const newAccount = await createAccount(email, hashedPasswd, miiName, pid, pnid);
+        const jwtToken = getAccountJwt(newAccount); // json web token token lol
+
+        // sends the token on success
+        res.json({
+            endpoint: "createAccount",
+            time: time,
+            error: 0,
+            result: {
+                success: 1,
+                token: jwtToken, 
+            },
+        });
+    } catch (e: unknown) {
+        // sends an error on error
+        res.status(500).json({
+            endpoint: "new",
+            time: time,
+            error: 1,
+            result: {
+                success: 0,
+            },
+        });
+        logger.error(e as string);
+    }
 });
 
 /**
- * [GET] /api/v1/act/login - Account login endpoint
+ * [GET] /api/v1/account/login - Account login endpoint
  * 
  * Endpoint to login to a Club Rosé account
  */
-router.put("/login", (req, res) => {
-    res.send({ success: 1 }).contentType("application/json");
+router.post("/login", async (req, res) => {
+    const email = req.body.email; // User emial
+    const passwd = req.body.password; // users password (unhashed)
+    const time = dayjs().format("HH:mm:ss"); // 24 hr time
+
+    if (!email || !passwd) {
```

Comment:
again, what's the purpose of the "not" operator on strings

Reply:
if they dont exist it will send a different response?